ArrayTypeStyle false positives in nested generics

Rule : [ArrayTypeStyle](https://checkstyle.sourceforge.io/config_misc.html#ArrayTypeStyle)

The `ArrayTypeStyle` seems to fail when checking the declaration inside nested generics. It works as expected with the 8.43 but fails with the [8.44](https://checkstyle.sourceforge.io/releasenotes.html#Release_8.44).

```java
// Test.java
public class Test {

    protected Pair<Integer, Object>[] values1; // OK
    protected Pair<Integer, Pair<String, Object>[]>[] values2; // OK
    protected Pair<Integer, Pair<String, Pair<String, Object>>[]>[] values3a; // KO!
    protected Pair<
        Integer,
        Pair<
            String,
            Pair<String, Object>
        >[]
    >[] values3b; // OK, but in fact it's the same as "values3a" with another format

    protected static class Pair<L, R> {
        protected L key;
        protected R value;
    }
}
```

```xml
<!-- checkstyle.xml -->
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN" "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <property name="severity" value="error"/>

    <module name="SuppressWithPlainTextCommentFilter"/>
    <property name="fileExtensions" value="java, properties, xml"/>
    <module name="TreeWalker">
        <module name="ArrayTypeStyle"/>
    </module>
</module>
```

```bash
$ java -jar checkstyle-8.44-all.jar -c checkstyle.xml Test.java
```
```
Starting audit...
[ERROR] /home/junior/Downloads/Test.java:5:66: Array brackets at illegal position. [ArrayTypeStyle]
Audit done.
Checkstyle ends with 1 errors
```

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/ArrayTypeStyleCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/ArrayTypeStyleCheck.java
@@ -146,7 +146,9 @@ public class ArrayTypeStyleCheck extends AbstractCheck {
 
             // force all methods to be Java style (see note in top Javadoc)
             final boolean isMethodViolation = isMethod && !isJavaStyle;
-            final boolean isVariableViolation = !isMethod && isJavaStyle != javaStyle;
+            final boolean isVariableViolation = !isMethod
+                    && isJavaStyle != javaStyle
+                    && typeAST.getType() != TokenTypes.TYPE_ARGUMENT;
 
             if (isMethodViolation || isVariableViolation) {
                 log(ast, MSG_KEY);

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/ArrayTypeStyleCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/ArrayTypeStyleCheckTest.java
@@ -84,6 +84,20 @@ public class ArrayTypeStyleCheckTest
     }
 
     @Test
+    public void testNestedGenerics()
+            throws Exception {
+        final DefaultConfiguration checkConfig =
+            createModuleConfig(ArrayTypeStyleCheck.class);
+        final String[] expected = {
+            "18:45: " + getCheckMessage(MSG_KEY),
+            "19:61: " + getCheckMessage(MSG_KEY),
+            "20:76: " + getCheckMessage(MSG_KEY),
+            "27:16: " + getCheckMessage(MSG_KEY),
+        };
+        verify(checkConfig, getPath("InputArrayTypeStyleNestedGenerics.java"), expected);
+    }
+
+    @Test
     public void testGetAcceptableTokens() {
         final int[] expected = {TokenTypes.ARRAY_DECLARATOR };
         final ArrayTypeStyleCheck check = new ArrayTypeStyleCheck();

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/arraytypestyle/InputArrayTypeStyleNestedGenerics.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/arraytypestyle/InputArrayTypeStyleNestedGenerics.java
@@ -1,0 +1,33 @@
+package com.puppycrawl.tools.checkstyle.checks.arraytypestyle;
+
+/* Config:
+ * default
+ */
+public class InputArrayTypeStyleNestedGenerics {
+    protected Pair<Integer, Object>[] values1; // ok
+    protected Pair<Integer, Pair<String, Object>[]>[] values2; // ok
+    protected Pair<Integer, Pair<String, Pair<String, Object>>[]>[] values3a; // ok
+    protected Pair<
+        Integer,
+        Pair<
+            String,
+            Pair<String, Object>
+        >[]
+    >[] values3b; // ok
+
+    protected Pair<Integer, Object> values1b[]; // violation
+    protected Pair<Integer, Pair<String, Object>[]> values2b[]; // violation
+    protected Pair<Integer, Pair<String, Pair<String, Object>>[]> values3ab[]; // violation
+    protected Pair<
+        Integer,
+        Pair<
+            String,
+            Pair<String, Object>
+        >[]
+    > values3bb[]; // violation
+
+    protected static class Pair<L, R> {
+        protected L key;
+        protected R value;
+    }
+}

